osu.Server.Spectator/Hubs/MultiplayerQueue.cs

```diff
@@ -301,5 +313,39 @@ private async Task updatePlaylistOrder(IDatabaseAccess db)
                 await hub.NotifyPlaylistItemChanged(room, item);
             }
         }
+
+        private IEnumerable<IEnumerable<T>> interleaveMany<T>(IEnumerable<IEnumerable<T>> collections)
```

Comment:
I'm having a very hard time understanding what this function is doing. Could you add some documentation explaining the overall purpose?

Reply:
From a read through of the code, I believe what this does is interleave a sequence of sequences in the order they're given. To give an actual example, let's say we have three enumerables:

```
A = [A1, A2, A3, A4]
B = [B1, B2]
C = [C1, C2, C3]
```

Then `interleaveMany(new[] { A, B, C })` would produce a single flattened enumerable containing

```
[A1, B1, C1, A2, B2, C2, A3, C3, A4]
```

I do agree that the implementation is pretty optimised, but... unorthodox, which isn't doing any favours wrt readability.

Reply:
I feel moving it to an extensions/utils class and providing a simplified explanation along with the example stated above would be enough to give this more sense IMO. The `RemoveAll` part in the method is a bit weird to read through as well, but if it's for optimisation, then I guess it's fine to be used as-is?

Reply:
I can understand the intention from the name of the function, but can agree it should be an isolated method, probably with a test showing it does what we expect it to.

Reply:
That's understandable, I can move it to an extension method with tests. The reason for the optimisation is because I'm wary of unlimited room sizes, but on another think over it it's not that big of a deal to just do `O(m*n)` iterations over the entire list, where `n` is the number of players and `m` is the maximum number of elements in any one player (capped at 3, currently). Thus I will be removing the optimisation.